compiler/daml-lf-verify/tests/DA/Daml/LF/Verify/Tests.hs

```diff
@@ -16,82 +17,256 @@ import DA.Bazel.Runfiles
 
 mainTest :: IO ()
 mainTest = defaultMain $ testGroup "DA.Daml.LF.Verify"
-  [ quickstartTests,
-    conditionalTests
+  [ quickstartTests
+  , generalTests
+  , conditionalTests
+  , recursionTests
+  , expectFail recursionTestFail
```

Comment:
For other reviewers: I’ve discussed this with @Gertjan423 and this looked like a very weird and complicated bug so given the limited time (i.e. no time at all) that he has left, we decided to mark it as an expected failure so we can merge this PR.